rotkehlchen/chain/ethereum/modules/the_graph/decoder.py

```diff
@@ -0,0 +1,179 @@
+TOPIC_STAKE_DELEGATED_LOCKED = b'\x040\x18?\x84\xd9\xc4P#\x86\xd4\x99\xda\x80eC\xde\xe1\xd9\xde\x83\xc0\x8b\x01\xe3\x9am!\x16\xc4;%'  # noqa: E501
+# example withdrawDelegated() call: https://etherscan.io/tx/0x49307751de5ba4cf98fccbdd1ab8387fd60a7ce120800212c216bf0a6a04acfa
+TOPIC_STAKE_DELEGATED_WITHDRAWN = b'\x1b.w7\xe0C\xc5\xcf\x1bX|\xebM\xae\xb7\xae\x00\x14\x8b\x9b\xda\x8fy\xf1\t>\xea\xd0\x8f\x14\x19R'  # noqa: E501
+
+
+class TheGraphDecoder(DecoderInterface):
+    def counterparties(self) -> list[CounterpartyDetails]:
+        return [THE_GRAPH_CPT_DETAILS]
+
+    def possible_events(self) -> DecoderEventMappingType:
+        return {CPT_THE_GRAPH: {
+            HistoryEventType.STAKING: {
+                HistoryEventSubType.DEPOSIT_ASSET: EventCategory.DEPOSIT,
+                HistoryEventSubType.REMOVE_ASSET: EventCategory.WITHDRAW,
+            },
+            HistoryEventType.INFORMATIONAL: {
+                HistoryEventSubType.NONE: EventCategory.INFORMATIONAL,
+            },
+        }}
+
+    def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
+        return {
+            CONTRACT_THE_GRAPH_STAKING: (self._decode_delegator_staking,),
+        }
+
+    # processes events from The Graph's staking contract
+    def _decode_delegator_staking(self, context: DecoderContext) -> DecodingOutput:
+        token = A_GRT.resolve_to_evm_token()
+        if context.tx_log.topics[0] == TOPIC_STAKE_DELEGATED:
+            return self._decode_stake_delegated(token, context)
+        elif context.tx_log.topics[0] == TOPIC_STAKE_DELEGATED_LOCKED:
+            return self._decode_stake_locked(token, context)
+        elif context.tx_log.topics[0] == TOPIC_STAKE_DELEGATED_WITHDRAWN:
+            return self._decode_stake_withdrawn(token, context)
+        return DEFAULT_DECODING_OUTPUT
+
+    # processes StakeDelegated event
+    def _decode_stake_delegated(
+            self,
+            token: 'EvmToken',
+            context: DecoderContext,
+    ) -> DecodingOutput:
+        delegator = hex_or_bytes_to_address(context.tx_log.topics[2])
+        if delegator is None or self.base.is_tracked(delegator) is False:
+            return DEFAULT_DECODING_OUTPUT
+        indexer = hex_or_bytes_to_address(context.tx_log.topics[1])
+        stake_amount = token_normalized_value(
+            token_amount=hex_or_bytes_to_int(context.tx_log.data[:32]),
+            token=token,
+        )
+        # find Transfer event with delegation tax (0.5% GRT burn) and persist the burnt amount
+        tokens_burnt = 0
+        for ctx_log in context.all_logs:
+            if (
+                ctx_log.topics[0] == TOPIC_TRANSFER and
+                hex_or_bytes_to_address(ctx_log.topics[1]) == CONTRACT_THE_GRAPH_STAKING and
+                hex_or_bytes_to_address(ctx_log.topics[2]) == ZERO_ADDRESS
+            ):
+                tokens_burnt = hex_or_bytes_to_int(ctx_log.data)
```

Comment:
This is good and will work for the example you have. The problem will come if someone writes a contract/script that does multiple delegations in 1 transaction.

Then this approach will find the last one. Notice you don't stop (do a `break`) in the for loop once you find it, thus you find the last burnt event.

Since you mentioned in discord that the burn % is fixed, then perhaps we can use this to our advantage in order to be even more accurate.

So find the burned event that matches the 0.5% of the staked amount, use that and then break out of the searching for loop

Reply:
[Updated](https://github.com/rotki/rotki/pull/6712/files#diff-95ffe39eb01c3fbcb9f750f20cf51d0e861a7be760f5e4921c42105b5e12e91dR102). I through through the possible variants on how to process it and figured that the subtraction is most okay-ish approach (and it also will work correct with multiple invocations of `StakeDelegated` decoding).
There's also not much reason to loop over `context.all_logs` and look for burn Transfers after all. Because the previous looping was through the logs - not decoded events accounting for burns. So we have nothing to re-purpose/override here and the search for the sake of validation doesn't have much meaning. We just need to add new events that will account for delegation taxes. So we just act on the assumption that every GRT transfer to the contract (that we didn't re-purpose/override yet) will create the relevant delegation tax spending event.